test/functional/wallet_tokens_change_authority.py

```diff
@@ -0,0 +1,183 @@
+            assert token_id is not None
+            assert err is None
+            self.log.info(f"new token id: {token_id}")
+
+            self.generate_block()
+            assert_in("Success", await wallet.sync())
+
+            assert_in("The transaction was submitted successfully", await wallet.mint_tokens(token_id, address, 10000))
+
+            self.generate_block()
+            assert_in("Success", await wallet.sync())
+
+            assert_in(f"{token_id} amount: 10000", await wallet.get_balance())
+
+            ## create a new account and send some tokens to it
+            await wallet.create_new_account()
+            await wallet.select_account(1)
+            new_acc_address = await wallet.new_address()
+
+            await wallet.select_account(DEFAULT_ACCOUNT_INDEX)
+            output = await wallet.send_tokens_to_address(token_id, new_acc_address, 10.01)
+            assert_in("The transaction was submitted successfully", output)
+            # send some coins for fees
+            assert_in("The transaction was submitted successfully", await wallet.send_to_address(new_acc_address, 600))
+
+            self.generate_block()
+            assert_in("Success", await wallet.sync())
+
+            ## check the new balance
+            assert_in(f"{token_id} amount: 9989.99", await wallet.get_balance())
+
+
+            assert_in("The transaction was submitted successfully", await wallet.change_token_authority(token_id, new_acc_address))
+
+            # randomly put the tx in a block or keep in mempool unconfirmed
+            self.generate_block()
+            assert_in("Success", await wallet.sync())
+
+
+            # try to mint, unmint, lock, freeze and unfreeze
+            assert_in("Cannot change a not owned token", await wallet.mint_tokens(token_id, address, 1))
+            assert_in("Cannot change a not owned token", await wallet.unmint_tokens(token_id, 1))
+            assert_in("Cannot change a not owned token", await wallet.lock_token_supply(token_id))
+            assert_in("Cannot change a not owned token", await wallet.freeze_token(token_id, 'unfreezable'))
+            assert_in("Cannot change a not owned token", await wallet.unfreeze_token(token_id))
+
+
+            await wallet.select_account(1)
+
+            # check that the other account is now the owner and can do anything
+            assert_in("The transaction was submitted successfully", await wallet.mint_tokens(token_id, address, 1))
+            assert_in("The transaction was submitted successfully", await wallet.unmint_tokens(token_id, 1))
+            assert_in("The transaction was submitted successfully", await wallet.lock_token_supply(token_id))
+            assert_in("The transaction was submitted successfully", await wallet.freeze_token(token_id, 'unfreezable'))
+            assert_in("The transaction was submitted successfully", await wallet.unfreeze_token(token_id))
+
+
+
+if __name__ == '__main__':
+    WalletTokens().main()
+
+
+
+
```

Comment:
Not that you have to change this now, maybe when you push next... but why too many lines at the end? One is more than enough :smile:

Reply:
I actually meant one `\n` at the end, not an extra empty line :D